src/app/shared/auth/auth.service.ts

```diff
@@ -81,11 +82,21 @@ export class AuthService implements OnDestroy {
         const refreshIn = 5000;
         return timer(Math.min(refreshIn, maxDelay), 1*MINUTES).pipe(map(() => auth));
       }),
-      switchMap(auth =>
-        this.authHttp.refreshAuth(auth)
-      )
-    ).subscribe(auth => {
-      this.status$.next(auth);
+      switchMap(auth => {
+        const tokenIsExpired = auth.expiration.valueOf() < Date.now();
+        if (tokenIsExpired) return of({ auth, tokenIsExpired }); // don't even try to refresh the token if it is expired
+        return this.authHttp.refreshAuth(auth).pipe(
+          catchError(err => {
+            // For any http error, ignore it since the token is valid. Another try will occur the next minute.
+            if (err instanceof HttpErrorResponse) return EMPTY;
+            else throw err; // rethrow any JS error to let our error monitor catch it.
+          }),
+          map(auth => ({ auth, tokenIsExpired })),
```

Comment:
For clarity, I would return `{ auth, tokenIsExpired: false }` instead (even `false` is the only possible value here) as there is not the expiration of the previous token that we return here (that may have become expired in the meantime) but the new one (which is non-expired hopefully).